app/javascript/components/enrollment/steps/Identification.js

```diff
@@ -227,112 +263,43 @@ class Identification extends React.Component {
                   <Form.Control
-                    isInvalid={this.state.errors['primary_language']}
                     size="lg"
                     className="form-square"
                     value={this.state.current.patient.primary_language || ''}
                     onChange={this.handleChange}
-                    as="input"
-                    list="languages"
-                  />
-                  <datalist id="languages">
+                    as="select">
                     <option></option>
-                    <option value="English">English*</option>
-                    <option value="Spanish">Spanish*</option>
-                    <option value="Spanish (Puerto Rican)">Spanish (Puerto Rican)*</option>
-                    <option value="French">French*</option>
-                    <option value="Somali">Somali*</option>
-                    <option value="Arabic">Arabic</option>
-                    <option value="Bengali">Bengali</option>
-                    <option value="Czech">Czech</option>
-                    <option value="Danish">Danish</option>
-                    <option value="German">German</option>
-                    <option value="Greek">Greek</option>
-                    <option value="Finnish">Finnish</option>
-                    <option value="Frysian">Frysian</option>
-                    <option value="Hindi">Hindi</option>
-                    <option value="Croatian">Croatian</option>
-                    <option value="Italian">Italian</option>
-                    <option value="Japanese">Japanese</option>
-                    <option value="Korean">Korean</option>
-                    <option value="Dutch">Dutch</option>
-                    <option value="Norwegian">Norwegian</option>
-                    <option value="Punjabi">Punjabi</option>
-                    <option value="Polish">Polish</option>
-                    <option value="Portuguese">Portuguese</option>
-                    <option value="Russian">Russian</option>
-                    <option value="Serbian">Serbian</option>
-                    <option value="Swedish">Swedish</option>
-                    <option value="Telegu">Telegu</option>
-                    <option value="Chinese">Chinese</option>
-                    <option value="Vietnamese">Vietnamese</option>
-                    <option value="Tagalog">Tagalog</option>
-                    <option value="Nepali">Nepali</option>
-                    <option value="Swahili">Swahili</option>
-                    <option value="Burmese">Burmese</option>
-                  </datalist>
-                  <Form.Control.Feedback className="d-block" type="invalid">
-                    {this.state.errors['primary_language']}
-                  </Form.Control.Feedback>
+                    {supportedLanguages.languages.map((language, languageIndex) => this.renderLanguageOption(language, languageIndex))}
```

Comment:
This feels slightly off to me. Why iterate over an object, calling a function for each iteration that returns JSX, when you could just call the function once, and have it iterate over the object and return all of the `<option.../>`s?

This is kind of a nitpick, but the logic just seems a little convoluted here